### PR TITLE
Add update user and update profile

### PR DESCRIPTION
**Pull request by yordis**

### CODE REVIEW

**Comment by yordis** [comment]
Hey @stueccles how are you testing this?

I ran the tests and all of them failed so I am probably missing something.

File: test/users/users_create_test.exs [hunk 1]
```diff
@@ -268,4 +268,50 @@ defmodule Okta.UsersCreateTest do
...
+    data = %{profile: %{login: "newlogin@acmec.com"}}
+
+    Okta.Tesla.Mock
+    |> Mox.expect(:call, fn
+      request, _opts ->
+        {:ok, expected_body} = Jason.encode(data)
+
+        assert expected_body == request.body
+        assert :put == request.method
+        assert [strict: false] = request.query
```
Comment (stueccles) [inline test/users/users_create_test.exs]: this test `assert [strict: false] = request.query` fails as the API call with `Okta.Users.update_user(client, user_id, data, [strict: true])` sets the query param `strict` to `true`

File: test/users/users_create_test.exs [hunk 2]
```diff
@@ -268,4 +268,50 @@ defmodule Okta.UsersCreateTest do
...
+        assert "#{base_url}/api/v1/users/#{user_id}" == request.url
+        {:ok, Tesla.Mock.json(%{}, status: 200)}
+    end)
+
+    assert {:ok, %{}, _env} =
+             Okta.Users.update_user(client, user_id, data, [strict: true])
```
Comment (stueccles) [inline test/users/users_create_test.exs]: should be `update_profile`

File: test/users/users_create_test.exs [hunk 3]
```diff
@@ -268,4 +268,50 @@ defmodule Okta.UsersCreateTest do
...
+    data = %{profile: %{login: "newlogin@acmec.com"}}
+
+    Okta.Tesla.Mock
+    |> Mox.expect(:call, fn
+      request, _opts ->
+        {:ok, expected_body} = Jason.encode(data)
+
+        assert expected_body == request.body
+        assert :put == request.method
+        assert [strict: false] = request.query
```
Comment (stueccles) [inline test/users/users_create_test.exs]: this will also fail if `strict: true` is used on the API call

File: test/users/users_create_test.exs [hunk 4]
```diff
@@ -268,4 +268,50 @@ defmodule Okta.UsersCreateTest do
...
+    user_id = "123"
+    data = %{profile: %{login: "newlogin@acmec.com"}}
+
+    Okta.Tesla.Mock
+    |> Mox.expect(:call, fn
+      request, _opts ->
+        {:ok, expected_body} = Jason.encode(data)
+
+        assert expected_body == request.body
+        assert :put == request.method
```
Comment (stueccles) [inline test/users/users_create_test.exs]: if `update_profile` will be `:post`

**Comment by stueccles** [comment]
hi @yordis im not sure why all the original tests fail for you. they pass for me with `mix test`. what kind of errors do you get?

your tests on this commit don't pass though, i left a couple of comments. i have checked against a live Okta API as well and the calls do work as expected.

**Comment by yordis** [comment]
@stueccles I had no clue why my tests didn't pass when I ran them before ................. 

>  "https://dev-000000.okta.com"

Is that a testing server btw? I am curious if there is a way to hit real Okta server in the test cases in testing environments.

**Comment by stueccles** [comment]
I don't think Okta have a test server. But you can sign up for a free account though to test

**Comment by yordis** [comment]
@stueccles btw could you make a release with this feature? I need these two functionalities 😄

**Comment by stueccles** [comment]
haha. sure. 0.1.5 is on hex now.

**Pull request merged**